wp-content/civi-extensions/goonjcustom/Civi/InductionService.php

```diff
@@ -187,16 +189,89 @@ private static function sendInductionEmail($contactId) {
       return FALSE;
     }
 
+    // Prepare email parameters.
     $emailParams = [
       'contact_id' => $contactId,
       'template_id' => $template['id'],
       'cc' => self::$volunteerInductionAssigneeEmail,
     ];
 
+    // Fetch contact's email.
+    $contacts = Contact::get(FALSE)
+      ->addSelect('email.email')
+      ->addJoin('Email AS email', 'LEFT')
+      ->addWhere('id', '=', $contactId)
+      ->execute()->single();
+
+    $email = $contacts['email.email'];
+
+    if (empty($email)) {
+      self::queueInductionEmail($emailParams);
+      return TRUE;
+    }
+
     civicrm_api3('Email', 'send', $emailParams);
     return TRUE;
   }
 
+  /**
+   * Queue the induction email to be processed later.
+   */
+  private static function queueInductionEmail($params) {
+    try {
+      $queue = \Civi::queue(\CRM_Goonjcustom_Engine::QUEUE_NAME, [
+        'type' => 'Sql',
+        'error' => 'abort',
+        'runner' => 'task',
+      ]);
+
+      $queue->createItem(new \CRM_Queue_Task(
+            [self::class, 'processQueuedInductionEmail'],
+            [$params]
+        ), [
+          'weight' => 1,
+        ]);
+
+      \Civi::log()->info('Induction email queued for contact', ['contactId' => $params['contact_id']]);
+    }
+    catch (\CRM_Core_Exception $ex) {
+      \Civi::log()->error('Failed to queue induction email due to CiviCRM error', [
+        'contactId' => $params['contact_id'],
+        'error' => $ex->getMessage(),
+      ]);
+    }
+    catch (\Exception $ex) {
+      \Civi::log()->error('Unexpected error while queueing induction email', [
+        'contactId' => $params['contact_id'],
```

Comment:
let's log all the params and not just contactId
```suggestion
        'paramas' => $params,
```